test/azure/common_test.go

```diff
@@ -15,6 +17,15 @@ import (
 func TestGetTargetAzureSubscription(t *testing.T) {
 	t.Parallel()
 
+	var id string
+	var exists bool
+
+	//Lookup ARM_SUBSCRIPTION_ID env variable, CI requires this value to run all test.
+	if id, exists = os.LookupEnv(azure.AzureSubscriptionID); !exists {
+		fmt.Printf("ARM_SUBSCRIPTION_ID environment variable not set")
+		id = ""
+	}
```

Comment:
Can this be simplified to:

```
require.NotEmpty(t, os.Getenv(azure.AzureSubscriptionID), "ARM_SUBSCRIPTION_ID environment variable not set")
```

This will fail the test if the environment variable is not set.

Reply:
This is a better solutoin.  I updated the test to include this check.  We still need to add a test to test throwing the exception.  I have someone on my team looking into that now and will submit in a seperate PR.